### PR TITLE
Ignore 'Caddyfile' (case-insensitive) in browse middleware

### PR DESCRIPTION
**Pull request by JosielSantos**
Simply ignore the "Caddyfile" in directory listing.

### CODE REVIEW

File: middleware/browse/browse.go [hunk 1]
```diff
@@ -92,189 +92,194 @@ func (l byTime) Less(i, j int) bool { return l.Items[i].ModTime.Before(l.Items[j
...
+                for _, indexName := range IndexPages {
+                        if name == indexName {
+                                return Listing{}, errors.New("Directory contains index file, not browsable!")
+                        }
+                }
+
+                // Ignore Caddyfile
+                if strings.ToLower(name)=="caddyfile" {
+                        continue
+                }
```
Comment (mholt) [inline middleware/browse/browse.go]: Ah, this must be the change. (Next time, run go fmt to keep the formatting consistent.)

**Comment by mholt** [comment]
Thanks for the contribution. I think we need to consider whether this is what we really want to do first. The bigger picture here is you want to hide or skip some files (or folders) that are within the directory you've exposed for browsing.

My first question is, can you get away without browsing a directory that potentially has sensitive information in it? Or move the sensitive files out of that directory? Second question is, what if the configuration file doesn't have the name of Caddyfile?

**Comment by mholt** [comment]
I think this is on the right track: to have the ability to hide certain files from the directory listings. I just want to make the solution more general so it can be more useful to more people. I think your feedback is really important in the process, so let me know what you think. :)

**Comment by pyed** [comment]
I don't see the need to hide certain files from a directory listing, you shouldn't expose any directories that contains sensitive information to begin with.

**Pull request closed without merging**